application/src/main/java/org/togetherjava/tjbot/commands/tags/TagManageCommand.java

```diff
@@ -156,25 +162,30 @@ private void rawTag(@NotNull SlashCommandEvent event) {
     private void createTag(@NotNull CommandInteraction event) {
         String content = Objects.requireNonNull(event.getOption(CONTENT_OPTION)).getAsString();
 
-        handleAction(TagStatus.NOT_EXISTS, id -> tagSystem.putTag(id, content), "created", event);
+        handleAction(TagStatus.NOT_EXISTS, id -> tagSystem.putTag(id, content), "created", event,
+                Subcommand.CREATE, content);
     }
 
     private void createTagWithMessage(@NotNull CommandInteraction event) {
-        handleActionWithMessage(TagStatus.NOT_EXISTS, tagSystem::putTag, "created", event);
+        handleActionWithMessage(TagStatus.NOT_EXISTS, tagSystem::putTag, "created", event,
+                Subcommand.CREATE_WITH_MESSAGE);
     }
 
     private void editTag(@NotNull CommandInteraction event) {
         String content = Objects.requireNonNull(event.getOption(CONTENT_OPTION)).getAsString();
 
-        handleAction(TagStatus.EXISTS, id -> tagSystem.putTag(id, content), "edited", event);
+        handleAction(TagStatus.EXISTS, id -> tagSystem.putTag(id, content), "edited", event,
+                Subcommand.EDIT, content);
     }
 
     private void editTagWithMessage(@NotNull CommandInteraction event) {
-        handleActionWithMessage(TagStatus.EXISTS, tagSystem::putTag, "edited", event);
+        handleActionWithMessage(TagStatus.EXISTS, tagSystem::putTag, "edited", event,
+                Subcommand.EDIT_WITH_MESSAGE);
     }
 
     private void deleteTag(@NotNull CommandInteraction event) {
-        handleAction(TagStatus.EXISTS, tagSystem::deleteTag, "deleted", event);
+        handleAction(TagStatus.EXISTS, tagSystem::deleteTag, "deleted", event, Subcommand.DELETE,
+                null);
```

Comment:
in fact, I could also move the lambda into the enum.

Reply:
You can. But maybe that overloads the enum a bit. I am unsure. Your choice.

Reply:
well, I've tried to pass it in the simplest way possible but it makes a very awkward and confusing code, so I won't pass the `FunctionalInterface`s through the enum. also, I think that passing the `TagStatus` will also lead to confusion because it plays a part in the tag managing process, and hiding it in the enum is not a good idea in my opinion. I'll just add the cosmetic `String actionVerb` to the enum.

Reply:
done.